routes/projects.js

```diff
@@ -1,9 +1,42 @@
-var express = require('express');
-var router = express.Router();
+const express = require('express')
+const router = express.Router()
+const getProjects = require('../xnat-apis/getProjects')
+const startSession = require('../xnat-apis/startSession')
+const deleteSession = require('../xnat-apis/deleteSession')
+const userAliasToken = require('../xnat-apis/userAliasToken')
 
 /* GET projects listing. */
-router.get('/', function(req, res, next) {
-  res.send('respond with a resource');
-});
+router.get('/:user', async function (req, res, next) {
```

Comment:
If you're pulling `user` from the keycloak token, do you need the `user` param here?

Reply:
Oh so my understanding was that the frontend will have a get projects button which will send in the user (either as `rdhar2` or as a JWT token from keycloak as param). So if that is not the case how will the frontend communicate the user id to the backend API?

Reply:
How will the keycloak send the token to backend?

Reply:
Oh! I think I misread something! Generally, authorization tokens come through in the authorization header, not as a parameter. I see that you're grabbing the user from the request parameters here:

```
const user = req.params.user
```

I think you can pull the user information from the `jwt` using the `user` part of the request.
https://www.npmjs.com/package/express-jwt#usage

Something like this:
```
const user = req.user.shortid
```

I see now that the `jwt` stuff has been removed from `app.js`. Are you trying to just get things working before doing authentication?

Reply:
> How will the keycloak send the token to backend?

Ahh, I understand! You don't! You use the keycloak token to get the `shortid` of the user. Once you have the `shortid` of the user, you use an `ADMINJSESSION` token to generate an alias token that can be used to get a `JSESSION` token for the user you want. Does that make sense?

Reply:
oh ok gotcha! I have some more basic questions mostly about keycloak, which seems new to me!
1. How do we get the keycloak token? Like is it from the frontend (send in as an auth header) or in backend?
2. Does `shortid` refer to the xnat username? because we need a username for xnat api `/data/services/tokens/issue/user/${user}`

Reply:
> oh ok gotcha! I have some more basic questions mostly about keycloak, which seems new to me!
> 
>     1. How do we get the keycloak token? Like is it from the frontend (send in as an auth header) or in backend?
> 
>     2. Does `shortid` refer to the xnat username? because we need a username for xnat api `/data/services/tokens/issue/user/${user}`

Got it!

1. The keycloak token comes from the frontend. The client requests a token from the keycloak server (completely different than `xnat` or this `xnat-portal` you're building). Then in all future requests to `xnat-portal` the frontend attaches the keycloak token as an authorization header
2. `shortid` is what the Identity Management team at Brown has labeled our computer logins. Mine is `broarr`, yours is `rdhar2`. `xnat` uses those same `shortid`s to authenticate individuals, but instead of using `shibboleth` or `keycloak`, `xnat` uses `ldap` to look up those usernames and passwords

Reply:
This makes sense!